### DISCUSSION TITLE
Not working post Quarkus version upgrade : Beans in dependent gradle modules not being created or not discoverable

### DISCUSSION THREAD

**Discussion by kirankolli-addepar** [quarkusio/quarkus#38924] [Q&A]
New Version : 3.2.10.Final 
Old Version : 2.16.5.Final

I have the following gradle setup : 

root
  -> module a
  -> module b

module b depends on module a and expresses this dependency via the build.gradle 
`dependencies {
  implementation project(':module a')
}`

Module B has a service class like so : 
````
@RequestScoped
public class Service{
  private final Repo repo;
 
  @Inject
  public Service(Repo repo){
    this.repo = repo;
  }
}
````

Module A has the needed class : 

````
@RequestScoped
public class Repo{
 // some methods
}
````

This setup was working fine in the old version but post upgrade Quarkus complains as follows : 

` ```
Unsatisfied dependency for type com.Service and qualifiers [@Default]
  - java member: com.Service():repo
````

Has anything changed in the versions that would need a change in the bean discovery ? 
The migration guides do not mention anything. 
Note : I did try using the quarkus cli to upgrade version and also the gradle task to upgrade but they always fail as they are unable to handle some custom extensions that our project uses and I am not aware of any flag that allows me to skip those extensions in the upgrade.

**Comment by kirankolli-addepar** [comment]
CC : @radcortez , @sberyozkin , @mkouba

[a comment by an automated account is not shown]

**Comment by mkouba** [comment]
I suppose there's a typo in the description and the `Repo` bean is in the module A?

In any case, I'm not aware of a breaking change in the [bean discovery](https://quarkus.io/guides/cdi-reference#bean_discovery).

Is module B the actual Quarkus app? 

In which mode (test, dev, production) do you get that error?

Ideally, we would need a minimal reproducer app and steps to reproduce the problem.

**Reply by kirankolli-addepar** [reply]
@mkouba its hard to make a reproducer since ours is a multi-module gradle project and there could very well be other nuances coming into play here. 
The only thing that caught my attention was that the bean which is not discoverable is `@RequestScoped` being injected into other service classes in another sub-module and teh service itself ends up in a Rest resource class that is annotated as : 

```
@RequestScoped
@Path("api/service/")
@Produces(MediaType.APPLICATION_JSON)
public class SomeResource {
```

do you think any of the above could be breaking bean injection ? if not, I will look for other potential breaks.

**Reply by mkouba** [reply]
I understand that's it hard to create a reproducer. Could you pls at least answer my questions from https://github.com/quarkusio/quarkus/discussions/38924#discussioncomment-8558876?

> do you think any of the above could be breaking bean injection ? if not, I will look for other potential breaks.

It's hard to guess without the code :shrug: .

**Comment by ricard2k** [comment]
I had a similar problem migrating from quarkus 2.16 to 3.2.11, but only in dev mode.

I recommend you to read and undertsand https://quarkus.io/guides/class-loading-reference and specifically the differences of the augmentation classloader.
